fix(orientation): refactor orientation detection and fix related issues

abbashosseinii opened the pull request:
This PR fixes several orientation-related issues by refactoring the orientation detection
  on Android.

  Changes:

   * Android: Replaced the BroadcastReceiver for ACTION_CONFIGURATION_CHANGED with an
     OrientationEventListener to get the physical device orientation instead of the UI
     orientation.
   * Overlay: The BarcodePainter now receives the deviceOrientation and adjusts the
     cameraPreviewSize accordingly, fixing the barcode overlay alignment on orientation
     change.
   * Version: Bumped version to 7.1.3.

  Related Issues:

  Fixes #1486
  Fixes #1474
  Fixes #1462

Commit:
feat(android): use OrientationEventListener for orientation

Commit:
fix(ios): use device orientation for video orientation

Commit:
fix(overlay): correct barcode overlay alignment on orientation change

Commit:
chore: bump version to 7.1.3

abbashosseinii committed:
Merge branch 'juliansteenbakker:develop' into fix/orientation-issues

navaronbracke commented on darwin/mobile_scanner/Sources/mobile_scanner/MobileScannerPlugin.swift:
```diff
@@ -290,42 +290,7 @@ public class MobileScannerPlugin: NSObject, FlutterPlugin, FlutterStreamHandler,
-        }
-
-        return videoOrientation
+        return UIDevice.current.orientation.videoOrientation
```
UIDeviceOrientation doesn't have a `videoOrientation` member?

https://developer.apple.com/documentation/uikit/uidevice/orientation

Or am I looking in the wrong place?

Isn't this the old switch case above? And what with the old comment:

```
// Get the orientation from the window scene if available
// When the app's orientation is fixed and the app orientation is actually different from the device orientation, it malfunctions.
```

does that still apply?

abbashosseinii replied:
Thanks for the thorough review and the LGTM—much appreciated!
Regarding your question on the iOS orientation handling: I've reverted those changes in [this commit](https://github.com/juliansteenbakker/mobile_scanner/pull/1552/commits/c13bb7178d0038c95ba5b8ba60bc74b879552215) because I don't have access to physical iOS devices to fully test the UIDeviceOrientation logic and ensure it aligns with the window scene fallback (or avoids the issues noted in the old comment). I was only able to verify the fixes on Android, so I'd prefer to hold off on iOS until I can test properly or get some guidance/eyes on it from the team.
If you'd like to take a stab at refining that part (or point me to the right approach for the videoOrientation mapping), I'm all ears—happy to iterate! Otherwise, let me know if this works for merging the Android-focused fixes as-is.
(Added a changelog entry for the Android refactors)

navaronbracke reviewed (commented):
Overall this is LGTM, except for one question on iOS.

Also, this needs a changelog entry.

FelipeABastos commented:
Nice one! Any estimate on when this will be merged?

juliansteenbakker commented:
@FelipeABastos I have a very busy week this week, so i'm aiming for next week.

abbashosseinii committed:
Revert "fix(ios): use device orientation for video orientation"

This reverts commit 4a3008aec729c4c208c524df897e62236e7a3595.

abbashosseinii committed:
fix: add missing deviceOrientation parameter to BarcodePainter tests

- Add flutter/services.dart import for DeviceOrientation enum
- Update all BarcodePainter constructor calls to include required deviceOrientation parameter
- Set deviceOrientation to DeviceOrientation.portraitUp for all test cases
- Fixes compilation errors after BarcodePainter constructor was updated to require deviceOrientation

All 6 tests now pass successfully.

abbashosseinii committed:
Revert "chore: bump version to 7.1.3"

This reverts commit 80d21954a8ff4a9471632d136d12b7636e656bdb.

abbashosseinii committed:
chore: update changelog with recent fixes for orientation handling

- Refactored Android orientation detection to use `OrientationEventListener`, improving accuracy during `MobileScanner` initialization.
- Updated `BarcodePainter` to dynamically adjust `cameraPreviewSize` based on `deviceOrientation`, resolving overlay misalignment during device rotations.

navaronbracke committed:
Remove header from CHANGELOG.md

As with official 1P plugins, these also don't use a header for the changelog, so I removed that here, too.

For example: 
https://github.com/flutter/packages/blob/main/packages/camera/camera/CHANGELOG.md

navaronbracke reviewed (approved):
LGTM!

navaronbracke reviewed (changes requested):
One more thing, it seems that one of the files was not formatted, using the 80 character line length limit:

```
   info • The line length exceeds the 80-character limit • lib/src/overlay/barcode_painter.dart:61:81 • lines_longer_than_80_chars
   info • The line length exceeds the 80-character limit • lib/src/overlay/barcode_painter.dart:73:81 • lines_longer_than_80_chars
```

abbashosseinii committed:
fix(lint): wrap long lines in barcode_painter.dart to comply with 80-char limit

abbashosseinii commented:
Also fixed a couple of long-line lints in the painter for good measure.

breezechime commented:
Is it surprising that the rotation direction of the phone is completely based on it, without even providing a parameter to lock the rotation?
